test/returns/lines.js

```diff
@@ -88,20 +102,14 @@ experiment('getMonths', () => {
 
   test('generates correct months', async () => {
     const months = getMonths('2018-01-01', '2018-12-31');
-    expect(months).to.equal([
-      { startDate: '2018-01-01', endDate: '2018-01-31', timePeriod: 'month' },
-      { startDate: '2018-02-01', endDate: '2018-02-28', timePeriod: 'month' },
-      { startDate: '2018-03-01', endDate: '2018-03-31', timePeriod: 'month' },
-      { startDate: '2018-04-01', endDate: '2018-04-30', timePeriod: 'month' },
-      { startDate: '2018-05-01', endDate: '2018-05-31', timePeriod: 'month' },
-      { startDate: '2018-06-01', endDate: '2018-06-30', timePeriod: 'month' },
-      { startDate: '2018-07-01', endDate: '2018-07-31', timePeriod: 'month' },
-      { startDate: '2018-08-01', endDate: '2018-08-31', timePeriod: 'month' },
-      { startDate: '2018-09-01', endDate: '2018-09-30', timePeriod: 'month' },
-      { startDate: '2018-10-01', endDate: '2018-10-31', timePeriod: 'month' },
-      { startDate: '2018-11-01', endDate: '2018-11-30', timePeriod: 'month' },
-      { startDate: '2018-12-01', endDate: '2018-12-31', timePeriod: 'month' }
-    ]);
+    expect(months).to.equal(expectedMonths);
+  });
+
+  test('generates correct months for partial period', async () => {
+    const months = getMonths('2018-01-01', '2018-12-18', true);
+    // last month should not be included for a partial period
+    expectedMonths.pop();
```

Comment:
Although pop works well here, it modifies the original array which could lead to unexpected results if tests are moved to a different order, run out of order, or further tests added.
It may be safer to make a fresh array along the lines of:
```
const expected = expectedMonths.slice(0, -1);
```